core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1637,6 +1650,26 @@ private ExprValue exprWeek(ExprValue date, ExprValue mode) {
         CalendarLookup.getWeekNumber(mode.integerValue(), date.dateValue()));
   }
 
+  /**
+   * Weekday implementation for ExprValue.
+   *
+   * @param date ExprValue of Date/Datetime/String/Timstamp type.
+   * @return ExprValue.
+   */
+  private ExprValue exprWeekday(ExprValue date) {
+    return new ExprIntegerValue(date.dateValue().getDayOfWeek().getValue() - 1);
+  }
+
+  /**
+   * weekday implementation for ExprValue when passing in an arguemt of type TIME.
+   *
+   * @param clock Current clock taken from function properties
+   * @return ExprValue.
+   */
+  private ExprValue weekdayToday(Clock clock) {
```

Comment:
Curious why you created a private function for a single call...?

Reply:
This.... is a good point. I did it this way because most of the other functions which exist follow a similar pattern, but it's definitely not needed. I will remove.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/234/commits/87c3f9ee724168af6276256415d875fd8511781a